server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -952,7 +953,12 @@ final ReaderContext createAndPutReaderContext(
                             + "This limit can be set by changing the ["
                             + MAX_OPEN_SCROLL_CONTEXT.getKey()
                             + "] setting."
-                    );
+                    ) {
+                        @Override
+                        public RestStatus status() {
+                            return RestStatus.BAD_REQUEST;
+                        }
```

Comment:
Do we think this is a `400` or a `429`?

Reply:
Interesting question. I've always interpreted 429 "too many requests" as a rate. 
> The HTTP 429 Too Many Requests response status code indicates the user has sent too many requests in a given amount of time

This is not an amount of time rate. It is just a count that within a query there are too many scroll requests, which you can see by the if-guard around this section:

```
        if (source.scriptFields() != null && source.size() != 0) {
            int maxAllowedScriptFields = searchExecutionContext.getIndexSettings().getMaxScriptFields();
            if (source.scriptFields().size() > maxAllowedScriptFields) {
```

Reply:
Got it. Changed.